src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+/**
+ * A subsystem that controls the intake on a robot.
+ */
+public class Intake extends SubsystemBase {
+  private final DoubleSolenoid solenoid;
+  private final CANSparkMax intakeMotor;
+
+  /**
+   * Constructs an Intake with a {@link CANSparkMax} at the given CAN ID and a
+   * {@link DoubleSolenoid} with the given forward and reverse channels.
+   *
+   * @param intakeMotorID The CAN ID of the intake motor
+   * @param forwardChannel The forward channel
+   * @param reverseChannel The reverse channel
+   */
+  public Intake(int intakeMotorID, int forwardChannel, int reverseChannel) {
+    this(intakeMotorID, MotorType.kBrushless, forwardChannel, reverseChannel);
+  }
+
+  /**
+   * Constructs an Intake with a {@link CANSparkMax} with the given CAN ID and motor type and a
+   * {@link DoubleSolenoid} with the given forward and reverse channels.
+   *
+   * @param intakeMotorID The CAN ID of the intake motor
+   * @param motorType The motor type of the intake motor
+   * @param forwardChannel The forward channel
+   * @param reverseChannel The reverse channel
+   */
+  public Intake(int intakeMotorID, MotorType motorType, int forwardChannel, int reverseChannel) {
+    solenoid = new DoubleSolenoid(PneumaticsModuleType.REVPH, forwardChannel, reverseChannel);
+    intakeMotor = new CANSparkMax(intakeMotorID, motorType);
+    intakeMotor.setIdleMode(IdleMode.kBrake);
+  }
+
+  /**
+   * Sets the intake motor speed.
+   * 
+   * @param intakeSpeed The speed of the intake motor
+   */
+  public void setSpinSpeed(double intakeSpeed) {
```

Comment:
Change "intakeSpeed" to "speed" or "newSpeed"

The method declaration should include a "throws" like this
```java
public void setSpinSpeed(double newSpeed) throws IllegalStateException {
```

Also, the javadoc should include a properly formatted "@throws" declaration